src/crypto/crypto_context.cc

```diff
@@ -658,33 +665,53 @@ void SecureContext::SetEngineKey(const FunctionCallbackInfo<Value>& args) {
 }
 #endif  // !OPENSSL_NO_ENGINE
 
+bool SecureContext::AddCert(BIOPointer&& bio) {
+  if (!bio) return false;
+  cert_.reset();
+  issuer_.reset();
+
+  if (!SSL_CTX_use_certificate_chain(
+          ctx_.get(), std::forward<BIOPointer>(bio), &cert_, &issuer_)) {
+    return false;
+  }
+
+  return true;
+}
+
 void SecureContext::SetCert(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
 
   SecureContext* sc;
   ASSIGN_OR_RETURN_UNWRAP(&sc, args.Holder());
 
-  CHECK_GE(args.Length(), 1);  // Certificate argument is mandator
+  CHECK_GE(args.Length(), 1);  // Certificate argument is mandatory
 
   BIOPointer bio(LoadBIO(env, args[0]));
   if (!bio)
     return;
 
-  sc->cert_.reset();
-  sc->issuer_.reset();
-
-  if (!SSL_CTX_use_certificate_chain(
-          sc->ctx_.get(),
-          std::move(bio),
-          &sc->cert_,
-          &sc->issuer_)) {
+  if (!sc->AddCert(std::move(bio))) {
     return ThrowCryptoError(
         env,
         ERR_get_error(),
         "SSL_CTX_use_certificate_chain");
   }
 }
 
+void SecureContext::SetCACert(const BIOPointer& bio) {
+  X509_STORE* cert_store = SSL_CTX_get_cert_store(ctx_.get());
+  while (X509* x509 = PEM_read_bio_X509_AUX(
+             bio.get(), nullptr, NoPasswordCallback, nullptr)) {
+    if (cert_store == root_cert_store) {
+      cert_store = NewRootCertStore();
+      SSL_CTX_set_cert_store(ctx_.get(), cert_store);
+    }
+    X509_STORE_add_cert(cert_store, x509);
+    SSL_CTX_add_client_CA(ctx_.get(), x509);
+    X509_free(x509);
```

Comment:
It'd be a little better to use `while (X509Pointer x509 = /* ... */) {` and avoid having to explicitly free here.

Reply:
Handled in #45912